crates/esthri/src/ops/sync.rs

```diff
@@ -445,10 +449,51 @@ where
 
     sync_tasks
         .buffer_unordered(task_count)
-        .try_collect()
+        .try_for_each_concurrent(task_count, |_| future::ready(Ok(())))
```

Comment:
Thinking that this will improve memory usage of the sync process since it does not collection result into a vector (which is immediately thrown away anyway).